### PR TITLE
Reduce overhead of imports when using AST interpreter

### PR DESCRIPTION
**Pull request by zooba**
In short, does a better job of scraping and importing modules until we really need to get them.

I'm expecting some test failures, but the run is going and I'll deal with them next week.

### CODE REVIEW

File: Python/Tests/DebuggerTests/AttachTests.cs [hunk 1]
```diff
@@ -777,6 +777,8 @@ void main()
         [TestMethod, Priority(0)]
         [TestCategory("10s")]
         public virtual async Task AttachWithOutputRedirection() {
+            Version.AssertInstalled();
```
Comment (huguesv) [inline Python/Tests/DebuggerTests/AttachTests.cs]: You should add this to CheckVersion() at the top of this class instead.

**Comment by huguesv** [comment]
👍

File: Python/Tests/DebuggerTests/AttachTests.cs (hunk 1 above)
Comment (zooba) [reply Python/Tests/DebuggerTests/AttachTests.cs]: Oddly enough, this didn't actually fix the problem... still looking into it.

**Comment by zooba** [comment]
@huguesv Made a couple more changes you may want to look over.

**Comment by huguesv** [comment]
👍

**Comment by zooba** [comment]
There are still some test regressions due to this change, but the fixes are going well beyond what this PR should do. Still working on them, but going to merge this and keep the fixes coming.

**Pull request merged**